File: test/llvm_test_code/linear_constant/global_13.cpp
int g = 0;

__attribute__((constructor)) void global_ctor() { g = 42; }

__attribute__((destructor)) void global_dtor() { g = 666; }

int foo(int x) {
  x = x + 1;
  return x;
}

int main() {
  int a = g;
  int b = foo(a);
  return g;
}

File: test/llvm_swift_test_code/linear_constant/global_13.swift
var g = 0

@_silgen_name("global_ctor") func globalCtor() {
    g = 42
}
@_silgen_name("global_dtor") func globalDtor() {
    g = 666
}

@main
struct MyMain {

  static func foo (_ x: Int)-> Int {
    var a = x
    a = a + 1
    return a
  }

    static func main() {
      var a = g
      var b = foo(a)
    }
}
